gremlin-go/driver/performance-tests/java/PerformanceTests.java

```diff
@@ -0,0 +1,200 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// CAUTION!!!
+// This file is intended for generating Performance test results only.
+// It is not to be used in a production setting.
+// It is being made available as a reference to the test results for
+// anyone that wants see the code used to obtain the performance metrics.
+
+package org.apache.tinkerpop.gremlin.driver;
+
+import lombok.AllArgsConstructor;
+import org.apache.tinkerpop.gremlin.driver.remote.DriverRemoteConnection;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__;
+import org.junit.Test;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.apache.tinkerpop.gremlin.process.traversal.AnonymousTraversalSource.traversal;
+
+class PerformanceTest {
+    final int SAMPLE_SIZE = 101;
+    final Cluster cluster = Cluster.build("localhost").port(8182).create();
+    final Client client = cluster.connect();
+    final Client g1client = client.alias("g");
+    final DriverRemoteConnection connection = DriverRemoteConnection.using(client, "ggrateful");
+    final GraphTraversalSource g = traversal().withRemote(connection);
+    final int REPEATS = 50;
+
+    public static GraphTraversal<?, ?> getTraversal(final String name, final GraphTraversalSource g) {
+        final Map<String, GraphTraversal<?, ?>> traversals = new HashMap<>();
+        traversals.put("getSongVertices", g.V().hasLabel("song"));
+        traversals.put("getNumVertices", g.V().count());
+        return traversals.get(name);
+
+    }
+
+    public static String[] getArgs(final int repeats) {
+        final String[] args = new String[repeats];
+        for (int i = 0; i < repeats; i++) {
+            args[i] = Integer.toString(i);
+        }
+        return args;
+    }
+
+    public static GraphTraversal<?, ?> getProjectTraversal(final GraphTraversalSource g, final int repeats, final String[] args) {
+        GraphTraversal<?, ?> traversal = g.V().project(args[0], Arrays.copyOfRange(args, 1, repeats));
+        for (int i = 0; i < repeats; i++) {
+            traversal = traversal.by(__.valueMap(true));
+        }
+        return traversal;
+    }
+
+    @Test
```

Comment:
Should we use main method instead of junit, so it can be compiled in a jar file?